src/gstreamer/gstkvssink.h

```diff
@@ -174,6 +176,7 @@ struct _KvsSinkCustomData {
     bool get_metrics;
     uint32_t frame_count;
     bool on_first_frame;
+    bool streamingStopped;
```

Comment:
Should this be atomic? I'm not sure but I suspect gst_kvs_sink_change_state() and gst_kvs_sink_handle_buffer() aren't guaranteed to be called from the same thread.

Reply:
In the stop path, I see it being on the same thread from the logs, but yes, I have made it atomic to be on the safe side, makes sense as well.